### PR TITLE
Better workspace/owner initialization

### PR DESCRIPTION
**Pull request by frascuchon**
Better data sync between `workspace` and `owner` attributes.

Also, force the workspace is present on a dataset instance creation

Note: Tests will be fixed in a future PR. This incremental PR is just to simplify the Reviewer's work.

### CODE REVIEW

File: src/argilla/server/daos/models/datasets.py [hunk 1]
```diff
@@ -36,24 +36,33 @@ class BaseDatasetDB(BaseModel):
...
 
-    @validator("workspace", pre=True, always=True)
-    def set_workspace_defaults(cls, value, values):
-        if value:
-            return value
-        else:
-            return values.get("owner")
+    @root_validator(pre=True)
+    def set_defaults(cls, values):
+        workspace = values.get("workspace")
```
Comment (jfcalvo) [inline src/argilla/server/daos/models/datasets.py]:
```suggestion
```

File: src/argilla/server/daos/models/datasets.py [hunk 2]
```diff
@@ -36,24 +36,33 @@ class BaseDatasetDB(BaseModel):
...
+        workspace = workspace or values.get("owner")
```
Comment (jfcalvo) [inline src/argilla/server/daos/models/datasets.py]:
```suggestion
        workspace = values.get("workspace") or values.get("owner")
```

File: tests/server/commons/test_records_dao.py [hunk 3]
```diff
@@ -26,6 +26,7 @@ def test_raise_proper_error():
         dao.search_records(
             dataset=BaseDatasetDB(
                 name="mock-notfound",
+                workspace="ee",
```
Comment (jfcalvo) [inline tests/server/commons/test_records_dao.py]: If `ee` is a random string maybe we should use more contextual strings like:

```suggestion
                workspace="workspace-name",
```

Correct me if I'm wrong.

**Review by jfcalvo — CHANGES_REQUESTED** [review]

File: tests/server/commons/test_records_dao.py (hunk 3 above)
Comment (frascuchon) [reply tests/server/commons/test_records_dao.py]: +1

**Review by jfcalvo — APPROVED** [review]

**Pull request merged**